### PR TITLE
Added kubectl gs template cluster doc for AWS CAPI

### PR DESCRIPTION
**Pull request by silenteh**
### What does this PR do?

First version of the `kubectl gs template cluster` doc for the `capa` provider (AWS CAPI)

### What does it look like?

Added additional TAB for AWS CAPI.  
Tried to make it clear, in the documentation that AWS CAPI != AWS

### Any background context you can provide?
Working on the `kubectl gs template cluster` docs for this issue:  
https://github.com/giantswarm/roadmap/issues/1435#issuecomment-1257871565

### What is needed from the reviewers?

Verify the changes are in line with the docs guidelines

### Have you maintained the front matter?
Bumped last_review_date

### CODE REVIEW

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 1]
```diff
@@ -84,16 +95,16 @@ The command to execute is `kubectl gs template cluster`.
...
-- `--service-priority` (optional) - [Service priority]({{< relref "/advanced/labelling-workload-clusters#service-priority" >}}) of the cluster (one of: `highest`, `medium`, or `lowest`; default: `highest`).
-- `--release-branch` (optional, AWS and Azure only) - The Giant Swarm [releases repository](https://github.com/giantswarm/releases) branch to use to look up the workload cluster release set via the `--release` flag (default: `master`).
+- `--pods-cidr` (optional and non CAPI AWS) - CIDR applied to the pods. If this isn't provided, the installation default will be applied.
+- `--label` (optional and non CAPI AWS) - workload cluster label in the form of `key=value`. Can be specified multiple times.
+- `--service-priority` (optional and non CAPI AWS) - [Service priority]({{< relref "/advanced/labelling-workload-clusters#service-priority" >}}) of the cluster (one of: `highest`, `medium`, or `lowest`; default: `highest`).
+- `--release-branch` (optional, non CAPI AWS and Azure only) - The Giant Swarm [releases repository](https://github.com/giantswarm/releases) branch to use to look up the workload cluster release set via the `--release` flag (default: `master`).
```
Comment (pipo02mix) [inline src/content/ui-api/kubectl-gs/template-cluster.md:107]:
```suggestion
- `--pods-cidr` (optional and non CAPI AWS) - CIDR applied to the pods. If this isn't provided, the installation default will be applied.
- `--label` (optional and non CAPI AWS) - workload cluster label in the form of `key=value`. Can be specified multiple times.
- `--service-priority` (optional and non CAPI AWS) - [Service priority]({{< relref "/advanced/labelling-workload-clusters#service-priority" >}}) of the cluster (one of: `highest`, `medium`, or `lowest`; default: `highest`).
- `--release-branch` (optional and non CAPI AWS and Azure only) - The Giant Swarm [releases repository](https://github.com/giantswarm/releases) branch to use to look up the workload cluster release set via the `--release` flag (default: `master`).
```

File: src/content/ui-api/management-api/crd/clusters.cluster.x-k8s.io.md [hunk 2]
```diff
@@ -20,13 +20,16 @@ crd:
   topics:
     - workloadcluster
   providers:
+    - aws CAPA
```
Comment (pipo02mix) [inline src/content/ui-api/management-api/crd/clusters.cluster.x-k8s.io.md:23]:
```suggestion
    - capa
```
I would use the same as the flag

**Review by pipo02mix — APPROVED** [review]
LGTM if Marian agrees

File: src/content/ui-api/kubectl-gs/template-cluster.md (hunk 1 above)
Comment (silenteh) [reply src/content/ui-api/kubectl-gs/template-cluster.md:107]: I have tried to set labels, pod-cidr and service-priority but I did not see those values applied to the resulting yaml manifests.  
This is why I removed those parameters, for now, until we investigate further.

**Comment by marians** [comment]
I'm addings suggestions to shorten and align these tab labels.

image

I think this is not the right place to have maturity indicators like "Beta" and "Alpha". And it should be consistent throughout the page. (Not blaming you Sebastian or anyone)

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 3]
```diff
@@ -29,6 +29,17 @@ For AWS (`--provider aws`):
 - [`G8sControlPlane`]({{< relref "/ui-api/management-api/crd/g8scontrolplanes.infrastructure.giantswarm.io.md" >}}) (API version `infrastructure.giantswarm.io/v1alpha3`) - specifies the control plane nodes
 - [`AWSControlPlane`]({{< relref "/ui-api/management-api/crd/awscontrolplanes.infrastructure.giantswarm.io.md" >}}) (API version `infrastructure.giantswarm.io/v1alpha3`) - specifies the control plane nodes with AWS-specific details
 
+{{< /tab >}}
+{{< tab id="flags-aws-capa" title="AWS CAPA (Beta)">}}
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:33]:
```suggestion
{{< tab id="flags-aws-capa" title="AWS CAPI">}}
```

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 4]
```diff
@@ -177,6 +192,20 @@ kubectl gs template cluster \
   --service-priority lowest
 ```
 
+{{< /tab >}}
+{{< tab id="command-examples-aws-capi" title="AWS CAPA">}}
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:196]:
```suggestion
{{< tab id="command-examples-aws-capi" title="AWS CAPI">}}
```

File: src/content/ui-api/management-api/crd/clusters.cluster.x-k8s.io.md (hunk 2 above)
Comment (marians) [reply src/content/ui-api/management-api/crd/clusters.cluster.x-k8s.io.md:23]: This page is generated and cannot be modified here.

**Comment by marians** [comment]
The page `src/content/ui-api/management-api/crd/clusters.cluster.x-k8s.io.md` cannot be edited manually, as it is generated and changes would be overwritten. To modify tags for a CRD, please modify https://github.com/giantswarm/docs/blob/main/scripts/update-crd-reference/config.yaml

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 5]
```diff
@@ -84,16 +95,16 @@ The command to execute is `kubectl gs template cluster`.
 
 It supports the following flags:
 
-- `--provider` - The infrastructure provider (one of: `aws`, `azure`, or `openstack`).
+- `--provider` - The infrastructure provider (one of: `capa`, `aws`, `azure`, or `openstack`).
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:98]: I suggest to sort the options alphabetically. Also `gcp` is missing.

```suggestion
- `--provider` - The infrastructure provider (one of: `aws`, `azure`, `capa`, `gcp`, or `openstack`).
```

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 6]
```diff
@@ -84,16 +95,16 @@ The command to execute is `kubectl gs template cluster`.
...
-- `--release` (AWS and Azure only) - Workload cluster release version.
+- `--release` (non CAPI AWS and Azure only) - Workload cluster release version.
   Can be retrieved with `kubectl get releases` for your installation.
 - `--description` (optional) - User-friendly description of the cluster's purpose.
-- `--pods-cidr` (optional) - CIDR applied to the pods. If this isn't provided, the installation default will be applied.
-- `--label` (optional) - workload cluster label in the form of `key=value`. Can be specified multiple times.
-- `--service-priority` (optional) - [Service priority]({{< relref "/advanced/labelling-workload-clusters#service-priority" >}}) of the cluster (one of: `highest`, `medium`, or `lowest`; default: `highest`).
-- `--release-branch` (optional, AWS and Azure only) - The Giant Swarm [releases repository](https://github.com/giantswarm/releases) branch to use to look up the workload cluster release set via the `--release` flag (default: `master`).
+- `--pods-cidr` (optional and non CAPI AWS) - CIDR applied to the pods. If this isn't provided, the installation default will be applied.
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:104]: "optional and non CAPI AWS" is not exactly clear to me. How about:

"optional, only non CAPI AWS"

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 7]
```diff
@@ -105,7 +116,11 @@ It supports the following flags:
   specify three distinct availability zones instead (AWS only). This can be done by separating AZ names with comma or using the flag
   three times with a single AZ name.
 
-### AWS specific
+### AWS specific (CAPI)
+
+ - `--name` must only contain alphanumeric characters, start with a letter, and be no longer than 5 characters in length
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:121]: Why 5? I remember we set a guideline to support a length of 10.

Comment (marians) [reply src/content/ui-api/kubectl-gs/template-cluster.md:121]: For reference: https://github.com/giantswarm/roadmap/issues/1017

File: src/content/ui-api/kubectl-gs/template-cluster.md (hunk 1 above)
Comment (silenteh) [reply src/content/ui-api/kubectl-gs/template-cluster.md:107]: Ah apologies - I have now realized this is about formatting

Comment (pipo02mix) [reply src/content/ui-api/kubectl-gs/template-cluster.md:107]: all good :)

File: src/content/ui-api/kubectl-gs/template-cluster.md (hunk 7 above)
Comment (silenteh) [reply src/content/ui-api/kubectl-gs/template-cluster.md:121]: As per our discussion `https://gigantic.slack.com/archives/C01F7T2MNRL/p1664271736738999` there is a validation currently happening in `kubectl gs` and a hidden flag to bypass the validation.  
This requires a new release of `Kubectl gs`  
So for now the documentation is correct in my opinion and we should address it once the changes are applied to `kubectl gs`  
I have opened an issue in Rainbow board: [ISSUE 1444](https://github.com/giantswarm/roadmap/issues/1444)

**Comment by silenteh** [comment]
@marians I believe all the issues were addressed.  
Let me know if you spot something else.  
Thanks a lot for the review!

**Comment by teemow** [comment]
Can this be merged now?

**Review by marians — APPROVED** [review]
Looks good from the docs side. Let's go.

File: src/content/ui-api/kubectl-gs/template-cluster.md (hunk 7 above)
Comment (marians) [reply src/content/ui-api/kubectl-gs/template-cluster.md:126]: ACK, this will be taken care of separately.

**Comment by marians** [comment]
@silenteh Do you still need anything, or can you merge?

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 8]
```diff
@@ -177,6 +197,20 @@ kubectl gs template cluster \
...
+{{< tab id="command-examples-aws-capi" title="AWS CAPI">}}
+
+Example command for an AWS CAPI cluster:
+
+```nohighlight
+kubectl gs template cluster \
+  --provider capa \  
+  --control-plane-az eu-central-1a \  
+  --description "Development Cluster" \  
+  --name dev01
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:210]: Backslash missing!

```suggestion
  --name dev01 \
```

File: src/content/ui-api/kubectl-gs/template-cluster.md [hunk 9]
```diff
@@ -361,6 +395,113 @@ spec:
...
+{{< /tab >}}
+{{< tab id="command-output-aws-capi" title="AWS CAPI">}}
+```yaml
+---
+apiVersion: v1
+data:
+  values: |
+    aws: {}
+    bastion: {}
+    clusterDescription: Test Cluster
```
Comment (marians) [inline src/content/ui-api/kubectl-gs/template-cluster.md:407]: Align with example command

```suggestion
    clusterDescription: Development Cluster
```

**Comment by silenteh** [comment]
We can merge 👍

**Comment by teemow** [comment]
@silenteh you merge. At giantswarm always the author merges. In case you didn't notice this yet :raised_hands:

**Comment by marians** [comment]
@paurosello As you mentioned you saw something missing in https://gigantic.slack.com/archives/C13MG9F0W/p1664534459348579 - Is this fine from your side?

**Comment by marians** [comment]
Can we please make sure this PR can get merged?

@silenteh As the author of the PR, it's up to you to merge once you are confident that the changes make sense, based on feedback.

I would like to proceed with restructuring the `kubectl gs template cluster` docs, and I would benefit from this being settled.

**Pull request merged**